### ISSUE TITLE
How can I use Doom implicit profiles to test Doom Emacs?

### ISSUE THREAD

**Issue by LemonBreezes** [doomemacs/doomemacs#7783]
### I confirm that...

- [X] I have searched the [issue tracker](https://github.com/doomemacs/doomemacs/issues), [documentation](https://docs.doomemacs.org), [FAQ](https://docs.doomemacs.org/-/faq), [Discourse](https://discourse.doomemacs.org), and [Google](https://google.com), in case this issue has already been reported/resolved.

- [X] I have read "[How to Debug Issues](https://doomemacs.org/d/how2debug)", and will use it to provide as much information about this issue as possible.

- [X] The issue can be reproduced on the **latest** available commit of Doom.

- [X] The issue can be reproduced on a stable release of Emacs, such as 27, 28, or 29. *(Unstable versions end in .50, .60, or .9x)*

### Expected behavior

Basically, I am wondering how the implicit profiles work. I created a `~/.config/doom/profiles/evil-mc-test` folder and ran `doom sync --profile evil-mc-test` but then when I ran `emacs --profile evil-mc-test`, it still seems to be loading my `config.el` and seeing my packages. I am not sure how this feature works.

### Current behavior

.

### Steps to reproduce

.

### System Information

N/A
